ord_schema/proto/reaction.proto

```diff
@@ -861,26 +860,136 @@ message ReactionProduct {
+  // of ostensibly pure compounds (e.g., HRMS) should have their results
+  // defined in the ReactionAnalysis message (with is_of_isolated_species set
+  // to True) but with a cross-reference to a product via a ProductMessage
+  // (e.g., of type IDENTITY for HRMS).
+  enum MeasurementType {
     UNSPECIFIED = 0;
     CUSTOM = 1;
-    EE = 2;  // Enantiomeric excess as a percentage (desired - undesired).
-    ER = 3;  // Enantiomeric ratio. (x:1) (desired / undesired).
-    DR = 4;  // Diasteromeric ratio. (x:1) (desired / undesired).
-    EZ = 5;  // Alkene geometry ratio. (x:1) (E / Z)
-    ZE = 6;  // Alkene geometry ratio. (x:1) (Z / E)
+    // An identity type indicates that the corresponding analysis was used
+    // to confirm the identity of the reported product.
+    IDENTITY = 2;
+    // Yields should only be reported as quantitative yields, e.g., measured
+    // using an isolated weight analysis, quantitative NMR with an internal
+    // standard, LC peak area with an internal standard and calibration curve
+    // obtained using an authentic standard. 
+    YIELD = 3;
+    // Selectivites include enantiomeric excess/ratios, diastereomic ratios,
+    // etc. The detailed type should be specified in selectivity_type.
+    SELECTIVITY = 4;
+    // The purity of the product mixture, normalized to all other species
+    // observed in the analysis.
+    PURITY = 5;
+    // Integrated peak area, e.g., as in an LC chromatogram.
+    AREA = 6;
+    // Raw counts, e.g., as in an EIC or TIC from a MS analysis. If an EIC,
+    // the m/z ratio(s) should be defined in the details.
+    COUNTS = 7;
+    // Intensity, e.g., of a TLC spot, ELSD, or GC FID.
+    INTENSITY = 8;
   }
-  SelectivityType type = 1;
-  string details = 2;
-  optional float value = 3;
-  // TODO(kearnes): What does precision mean in this context?
-  optional float precision = 4;
+  MeasurementType type = 2;
+  // Additional information about the measurement and, in particular, how it
+  // was calculated if that is ambiguous given the analysis type.
+  string details = 3;
+  // Whether an internal standard was used for this measurement to normalize
+  // it, e.g., for quantification of yield or relative peak area. If this
+  // field is true, the value field should contain the normalized quantity.
+  optional bool uses_internal_standard = 4;
+  // Whether an authentic sample of a material was used for callibrating
+  // an analytical technique, e.g., for product identification by LCMS.
+  // "Authentic" refers to a sample of a compound whose structure has been
+  // proven by the accepted means (1H NMR, 13C NMR, IR, HRMS), and is now
+  // being used as a comparison for other analytical methods.
+  optional bool uses_authentic_standard = 5;
+  // Details about the authentic standard used, optionally including
+  // information about its precise salt form and source/vendor.
+  Compound authentic_standard = 6;
```

Comment:
Since we have this `Compound` here, we could just check whether it is defined instead of having the additional `bool`. Or if the bool is handy we should make sure to check consistency in validations.

Reply:
We forgot to discuss this...

Reply:
I think the bool is handy, and there are cases where an internal standard is used but there might be some ambiguity in the publication or lack of details. I think that we can add a validation error if the compound is defined but the bool isn't True and a validation warning if the bool is True but the compound is not defined